### DISCUSSION TITLE
Width / positioning of floating element within tagged "input" using useListNavigation

### DISCUSSION THREAD

**Discussion by what-is-a-crow** [floating-ui/floating-ui#1787] [React] [answered]
We are building a multi-select autocomplete using `useListNavigation` that displays the selections as interactive tags. To do this, we have an unstyled text input inside of a div that is styled to look like a text input. The functionality so far is perfect; however, we're not sure how to work with the width / positioning of the listbox given the (real) text input _must_ be the reference but does not represent the width of the entire control. We can use `size` and set a `minWidth` based on an element that _does_ represent the control's width, but the listbox still positions itself relative to the (real) text input, leading to proper size but misalignment.

Any creative ideas on how to manage this?

**Comment by atomiks** [comment, marked as answer]
You can use a virtual element that copies over the real element's (the outer container/whole control) positioning data:  https://github.com/floating-ui/floating-ui/discussions/1759#discussioncomment-3040117
